### ISSUE TITLE
'Goto Deklaration' does not work in JSF

### ISSUE THREAD

**Issue by helper14** [apache/netbeans#4160]
### Apache NetBeans version

Apache NetBeans 15

### What happened

'Goto Deklaration' does not work in JSF

### How to reproduce

Rightclick on a method of a managed bean in a JSF and click 'Navigate -> Go to Declaration' ... nothing happens.

### Did this work correctly in an earlier version?

No

### Operating System

Windows 10 version 10.0 running on amd64; Cp1252; en_US (nb)

### JDK

17.0.3; OpenJDK 64-Bit Server VM 17.0.3+7

### Apache NetBeans packaging

Apache NetBeans provided installer

### Are you willing to submit a pull request?

No

### Code of Conduct

Yes

**Comment by triviFab** [comment]
it works correctly on Netbeans 12.1

**Comment by helper14** [comment]
Hello triviFab,
thank you for your reply! I tested it again on Netbeans 14 and 17.0.3; OpenJDK: Still does not work for me. 

I attached a minimal test project. If you open index.xhtml, right klick on 'myManagedBean' and kick 'Navigate > Goto Declaration' ... nothing happens. The same is for the 'text'-part of 'myManagedBean.text'.

[GotoProjectDeclaration.zip](https://github.com/apache/netbeans/files/9988623/GotoProjectDeclaration.zip)

**Comment by triviFab** [comment]
Hello helper14!

I pesuaded it doesn't work, I still struggle with it as well !

Last version I was able to use is, as I wrote before, 12.1

**Comment by matthiasblaesing** [comment]
Tested on a recent NB 19 and found, that with the provided test project, "Go to declaration" now works with CDI beans in the jakarata namespace.

**Comment by helper14** [comment]
Hello Matthias Bläsing,
I updated to Netbeans 19 with openjdk-19.0.1 (still on Windows 10) but still no luck: Nothing happens if I click the 'Go to Declaration' menu. An screenshot is attended. Anything I can do or try ...?

Regards!
 
![netbeans-screen](https://github.com/apache/netbeans/assets/83238864/63bffed9-608e-43fe-b498-3377a9c7aac8)

**Comment by triviFab** [comment]
@helper14 Any "unexpeted exception" notified (bubble in the lower side of the window) when you perform the Go to Declaration?

**Comment by helper14** [comment]
... no bubble is shown und there is no error in the notifications.

**Comment by matthiasblaesing** [comment]
1. Run with a clean userdir. For example I just ran `~/bin/netbeans-19/bin/netbeans --userdir x --cachedir x`. That command starts my NetBeans installation, with the folder `x` as the user and cachedir. It ensures, that no old data can interfere.
2. Don't mess with with settings. In your toolbar it is visible, that you disable IO checks, don't!

I tested again with NB19 and a clean userdir and goto declaration still works.
